server/memphis_sdk_handlers.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2021-2022 The Memphis Authors
+// Licensed under the Apache License, Version 2.0 (the “License”);
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an “AS IS” BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"encoding/json"
+)
+
+type simplifiedMsgHandler func(string, string, []byte)
+
+type createFactoryRequest struct {
+	Username    string `json:"username"`
+	FactoryName string `json:"factory_name"`
+	FactoryDesc string `json:"factory_description"`
+}
+
+type destroyFactoryRequest struct {
+	FactoryName string `json:"factory_name"`
+}
+
+type createStationRequest struct {
+	StationName       string `json:"name"`
+	FactoryName       string `json:"factory_name"`
+	RetentionType     string `json:"retention_type"`
+	RetentionValue    int    `json:"retention_value"`
+	StorageType       string `json:"storage_type"`
+	Replicas          int    `json:"replicas"`
+	DedupEnabled      bool   `json:"dedup_enabled"`
+	DedupWindowMillis int    `json:"dedup_window_in_ms"`
+	Username          string `json:"username"`
+}
+
+type destroyStationRequest struct {
+	StationName string `json:"station_name"`
+}
+
+type createProducerRequest struct {
+	Name         string `json:"name"`
+	StationName  string `json:"station_name"`
+	ConnectionId string `json:"connection_id"`
+	ProducerType string `json:"producer_type"`
+	Username     string `json:"username"`
+}
+
+type destroyProducerRequest struct {
+	StationName  string `json:"station_name"`
+	ProducerName string `json:"name"`
+	Username     string `json:"username"`
+}
+
+type createConsumerRequest struct {
+	Name             string `json:"name"`
+	StationName      string `json:"station_name"`
+	ConnectionId     string `json:"connection_id"`
+	ConsumerType     string `json:"consumer_type"`
+	ConsumerGroup    string `json:"consumers_group"`
+	MaxAckTimeMillis int    `json:"max_ack_time_ms"`
+	MaxMsgDeliveries int    `json:"max_msg_deliveries"`
+	Username         string `json:"username"`
+}
+
+type destroyConsumerRequest struct {
+	StationName  string `json:"station_name"`
+	ConsumerName string `json:"name"`
+	Username     string `json:"username"`
+}
+
+func (s *Server) initialiseSDKHandlers() {
+	// factories
+	s.subscribeOnGlobalAcc("$memphis_factory_creations",
+		"memphis_factory_creations_subscription",
```

Comment:
in case the second parameter is queue group all fine, otherwise we need to ensure this subscription is working as a group